### PR TITLE
Track SKU over ID for product

### PR DESCRIPTION
**Pull request by vincentchio**
## Unit testing
run `make test` keeps reporting `--harmony-generator` error. One 💩 way of getting over it is to remove the `--harmony-generator` from the `args` variable in the `node_modules/duo-test/bin/duo-test` file.

## Todo
- [x] unit test

### CODE REVIEW

**Comment by christianblais** [comment]
@peterjm for review

File: lib/index.js [hunk 1]
```diff
@@ -720,7 +720,7 @@ GA.prototype.refundedOrderEnhanced = function(track) {
   each(products, function(product) {
     var track = new Track({ properties: product });
     window.ga('ec:addProduct', {
-      id: track.id() || track.sku(),
+      id: track.sku() || track.id(),
```
Comment (peterjm) [inline lib/index.js]: This is a change in current behaviour, right? The code change makes sense, but what's the impact to merchants who were expecting the old behaviour?

**Comment by peterjm** [comment]
Code changes look fine, although Vincent mentioned it's possible to test this functionality?

Also a question about impact on existing merchants.

**Comment by christianblais** [comment]
Technically it won't be a problem since `id` is never sent. That's why we need this to be merged first, then we'll be able to merge the other PR where we start sending the `id` as well.

**Comment by peterjm** [comment]
Ah, cool.

:+1:

**Comment by vincentchio** [comment]
Added unit test to cover the new change. It is surprising that the GA integration does not have any unit test coverage to factor in the `id` tracking property. It seems to always assume the use of SKU which makes this change much more rest-assuring. The added unit test is pretty 💩 because the original unit test is full of code duplication.

**Comment by mkcny** [comment]
LGTM 👍

**Pull request merged**